pkg/resources/job.go

```diff
@@ -360,6 +371,16 @@ func resourceJobUpdate(ctx context.Context, d *schema.ResourceData, m interface{
 			deferringJobId := d.Get("deferring_job_id").(int)
 			job.Deferring_Job_Id = &deferringJobId
 		}
+		// If self_deferring has been toggled to true, set deferring_job_id as own ID
+		// Otherwise, set it back to what deferring_job_id specifies it to be
+		if d.HasChange("self_deferring") {
+			if d.Get("self_deferring") == true {
+				job.Deferring_Job_Id = job.ID
```

Comment:
maybe set this to `jobId` instead? the [tests](https://github.com/GtheSheep/terraform-provider-dbt-cloud/actions/runs/3178279205/jobs/5179609407) don't like some kind of ID

Reply:
I don't think the acceptance tests actually cover `resourceJobUpdate` do they? I might be mistaken. Based on the error output I see in the tests, it seems like only the following methods were called:
```
resourceJobCreate -> Client.CreateJob -> Client.UpdateJob
```
These lines: https://github.com/GtheSheep/terraform-provider-dbt-cloud/pull/86/files/2f9b684d9d1af9832509097c8a6a49c45ac14f3a#diff-35de52af826f07e23da639c22db938021a3e90c0eaa0b4cf27fc4f83c2d295a0R182-R184
seem to be the ones in this call stack that have to do with passing in a job's own ID into an update. Not sure what the issue is, but I'll get a dbt trial account and test there.

Reply:
I'll also add an additional acceptance test step to test resource update

Reply:
from [here](https://github.com/GtheSheep/terraform-provider-dbt-cloud/blob/36361f8381fa5d6bb8f9d732984dba18455eb7ac/pkg/resources/job_acceptance_test.go#L34) down is essentially testing updating a job resource, just doesn't check the values actually which it probably should.

But yeah, it's those update lines in the create that it's not liking, probs worth printing out the ID it's trying to update and the IDs used in that block as it's not matching somewhere 😬 seems like a minor issue tho tbf